File: lib/intern/cover.c
/***********************************************************
 * This file is part of glyr
 * + a commnadline tool and library to download various sort of musicrelated metadata.
 * + Copyright (C) [2011]  [Christopher Pahl]
 * + Hosted at: https://github.com/sahib/glyr
 *
 * glyr is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * glyr is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with glyr. If not, see <http://www.gnu.org/licenses/>.
 **************************************************************/

#include "../core.h"
#include "../stringlib.h"
#include "generic.h"

/* ------------------------------------- */

static GList * factory(GlyrQuery * s, GList * list, gboolean * stop_me)
{
    return generic_img_finalizer(s,list,stop_me,GLYR_TYPE_COVER);
}

/* ------------------------------------- */

bool vdt_cover(GlyrQuery * settings)
{
    if(settings && settings->artist && settings->album)
    {
		if(settings->artist[0] && settings->album[0])
		{
				/* validate size */
				if(settings->img_min_size <= 0)
					settings->img_min_size = -1;

				if(settings->img_max_size <= 0)
					settings->img_max_size = -1;

				return TRUE;
		}
    }
    glyr_message(2,settings,"Artist and Album is needed to retrieve coverart.\n");
    return FALSE;
}

/* ------------------------------------- */

/* PlugStruct */
MetaDataFetcher glyrFetcher_cover =
{
    .name = "cover",
    .type = GLYR_GET_COVERART,
    .validate  = vdt_cover,
    .full_data = FALSE,
    .init    = NULL,
    .destroy = NULL,
    .finalize = factory,
    .default_parallel = 3
};

/* ------------------------------------- */

File: spec/provider/tests/cover.py
#################################################################
# This file is part of glyr
# + a command-line tool and library to download various sort of music related metadata.
# + Copyright (C) [2011-2012]  [Christopher Pahl]
# + Hosted at: https://github.com/sahib/glyr
#
# glyr is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# glyr is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with glyr. If not, see <http://www.gnu.org/licenses/>.
#################################################################
#!/usr/bin/env python
# encoding: utf-8

from tests.__common__ import *

not_found_options = {
        'get_type': 'cover',
        'artist': 'Adele',
        'album':  '39',
        'title': 'Accidentally given'
        }


TESTCASES = [{
    # {{{
    'name': 'lyricswiki',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Axxis',
            'album': 'Access All Areas',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }],
    }, {
    # }}}
    # {{{
    'name': 'albumart',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Pussycat Dolls',
            'album': 'Dolls Domination',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'discogs',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Nirvana',
            'album': 'Smells like Nirvana',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'lastfm',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Alestorm',
            'album': 'Black Sails at Midgnight',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'google',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Amon Amarth',
            'album': 'Fate of Norns',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'picsearch',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Amon Amarth',
            'album': 'Fate of Norns',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'rhapsody',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'knorkator',
            'album': 'hasenchartbreaker',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'jamendo',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Diablo Swing Orchestra',
            'album': 'The Butcher\'s Ballroom',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'musicbrainz',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Akrea',
            'album': 'Lebenslinie',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }, {
    # }}}
    # {{{
    'name': 'musicbrainz',
    'data': [{
        'options': {
            'get_type': 'cover',
            'artist': 'Akrea',
            'album': 'Lebenslinie',
            },
        'expect': len_greater_0
        }, {
        'options': not_found_options,
        'expect': len_equal_0
        }]
    }
    # }}}
    ]
